src/System.Security.Cryptography.Primitives/src/System/Security/Cryptography/HashAlgorithm.cs

```diff
@@ -68,20 +68,40 @@ public byte[] ComputeHash(Stream inputStream)
             // Default the buffer size to 4K.
             byte[] buffer = new byte[4096];
             int bytesRead;
+            int bytesRead_firstBlock = 0;
             while ((bytesRead = inputStream.Read(buffer, 0, buffer.Length)) > 0)
             {
+                if (bytesRead_firstBlock == 0)
+                {
+                    bytesRead_firstBlock = bytesRead;
+                }
                 HashCore(buffer, 0, bytesRead);
             }
-            return CaptureHashCodeAndReinitialize();
+            return CaptureHashCodeAndReinitialize(bytesRead_firstBlock);
         }
 
-        private byte[] CaptureHashCodeAndReinitialize()
+        private byte[] CaptureHashCodeAndReinitialize(int inputCount)
```

Comment:
Why do we process the input count after doing all the work?  I'm a bit confused...